Updates to EnergyCal and GROOT

r3dunlop opened the pull request:
Use 'p' to fit a TPeak...It's pretty boss....

pcbend commented:
I was  going to use p for 2d project...  how would you fell about switching
to f for find or s for search?  or i need to find a new letter for project!

On Fri, Apr 24, 2015 at 2:57 PM, Ryan Dunlop <notifications@github.com>
wrote:

> Use 'p' to fit a TPeak...It's pretty boss....
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/pcbend/GRSISort/pull/286
> Commit Summary
>
>    - Nucleus loads into EnergyCal
>    - Can change energy calibration points
>    - Added TPeak to Groot
>
> File Changes
>
>    - *M* include/GCanvas.h
>    <https://github.com/pcbend/GRSISort/pull/286/files#diff-0> (2)
>    - *M* include/TCal.h
>    <https://github.com/pcbend/GRSISort/pull/286/files#diff-1> (3)
>    - *M* include/TEnergyCal.h
>    <https://github.com/pcbend/GRSISort/pull/286/files#diff-2> (7)
>    - *M* include/TNucleus.h
>    <https://github.com/pcbend/GRSISort/pull/286/files#diff-3> (2)
>    - *M* libraries/GROOT/GCanvas.cxx
>    <https://github.com/pcbend/GRSISort/pull/286/files#diff-4> (72)
>    - *M* libraries/TGRSIAnalysis/TCal/TCal.cxx
>    <https://github.com/pcbend/GRSISort/pull/286/files#diff-5> (16)
>    - *M* libraries/TGRSIAnalysis/TCal/TEnergyCal.cxx
>    <https://github.com/pcbend/GRSISort/pull/286/files#diff-6> (47)
>
> Patch Links:
>
>    - https://github.com/pcbend/GRSISort/pull/286.patch
>    - https://github.com/pcbend/GRSISort/pull/286.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pcbend/GRSISort/pull/286>.
>

r3dunlop commented:
f for fit is fine to me. I’ll do it on my end.

> On Apr 24, 2015, at 3:19 PM, pcbend <notifications@github.com> wrote:
> 
> I was going to use p for 2d project... how would you fell about switching
> to f for find or s for search? or i need to find a new letter for project!
> 
> On Fri, Apr 24, 2015 at 2:57 PM, Ryan Dunlop <notifications@github.com>
> wrote:
> 
> > Use 'p' to fit a TPeak...It's pretty boss....
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/pcbend/GRSISort/pull/286
> > Commit Summary
> >
> > - Nucleus loads into EnergyCal
> > - Can change energy calibration points
> > - Added TPeak to Groot
> >
> > File Changes
> >
> > - *M* include/GCanvas.h
> > <https://github.com/pcbend/GRSISort/pull/286/files#diff-0> (2)
> > - *M* include/TCal.h
> > <https://github.com/pcbend/GRSISort/pull/286/files#diff-1> (3)
> > - *M* include/TEnergyCal.h
> > <https://github.com/pcbend/GRSISort/pull/286/files#diff-2> (7)
> > - *M* include/TNucleus.h
> > <https://github.com/pcbend/GRSISort/pull/286/files#diff-3> (2)
> > - *M* libraries/GROOT/GCanvas.cxx
> > <https://github.com/pcbend/GRSISort/pull/286/files#diff-4> (72)
> > - *M* libraries/TGRSIAnalysis/TCal/TCal.cxx
> > <https://github.com/pcbend/GRSISort/pull/286/files#diff-5> (16)
> > - *M* libraries/TGRSIAnalysis/TCal/TEnergyCal.cxx
> > <https://github.com/pcbend/GRSISort/pull/286/files#diff-6> (47)
> >
> > Patch Links:
> >
> > - https://github.com/pcbend/GRSISort/pull/286.patch
> > - https://github.com/pcbend/GRSISort/pull/286.diff
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/pcbend/GRSISort/pull/286>.
> >
> —
> Reply to this email directly or view it on GitHub <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040476>.
>

pcbend commented:
ok, awesome.

On Fri, Apr 24, 2015 at 3:20 PM, Ryan Dunlop <notifications@github.com>
wrote:

> f for fit is fine to me. I’ll do it on my end.
>
> > On Apr 24, 2015, at 3:19 PM, pcbend <notifications@github.com> wrote:
> >
> > I was going to use p for 2d project... how would you fell about switching
> > to f for find or s for search? or i need to find a new letter for
> project!
> >
> > On Fri, Apr 24, 2015 at 2:57 PM, Ryan Dunlop <notifications@github.com>
> > wrote:
> >
> > > Use 'p' to fit a TPeak...It's pretty boss....
> > > ------------------------------
> > > You can view, comment on, or merge this pull request online at:
> > >
> > > https://github.com/pcbend/GRSISort/pull/286
> > > Commit Summary
> > >
> > > - Nucleus loads into EnergyCal
> > > - Can change energy calibration points
> > > - Added TPeak to Groot
> > >
> > > File Changes
> > >
> > > - *M* include/GCanvas.h
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-0> (2)
> > > - *M* include/TCal.h
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-1> (3)
> > > - *M* include/TEnergyCal.h
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-2> (7)
> > > - *M* include/TNucleus.h
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-3> (2)
> > > - *M* libraries/GROOT/GCanvas.cxx
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-4> (72)
> > > - *M* libraries/TGRSIAnalysis/TCal/TCal.cxx
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-5> (16)
> > > - *M* libraries/TGRSIAnalysis/TCal/TEnergyCal.cxx
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-6> (47)
> > >
> > > Patch Links:
> > >
> > > - https://github.com/pcbend/GRSISort/pull/286.patch
> > > - https://github.com/pcbend/GRSISort/pull/286.diff
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/pcbend/GRSISort/pull/286>.
> > >
> > —
> > Reply to this email directly or view it on GitHub <
> https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040476>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040661>.
>

pcbend commented:
We need to make a new rule; no adding keyboard commands with out updating
the wiki!   :)

https://github.com/pcbend/GRSISort/wiki/Interactive-Analysis

I even made a link on the main page.

On Fri, Apr 24, 2015 at 3:20 PM, Ryan Dunlop <notifications@github.com>
wrote:

> f for fit is fine to me. I’ll do it on my end.
>
> > On Apr 24, 2015, at 3:19 PM, pcbend <notifications@github.com> wrote:
> >
> > I was going to use p for 2d project... how would you fell about switching
> > to f for find or s for search? or i need to find a new letter for
> project!
> >
> > On Fri, Apr 24, 2015 at 2:57 PM, Ryan Dunlop <notifications@github.com>
> > wrote:
> >
> > > Use 'p' to fit a TPeak...It's pretty boss....
> > > ------------------------------
> > > You can view, comment on, or merge this pull request online at:
> > >
> > > https://github.com/pcbend/GRSISort/pull/286
> > > Commit Summary
> > >
> > > - Nucleus loads into EnergyCal
> > > - Can change energy calibration points
> > > - Added TPeak to Groot
> > >
> > > File Changes
> > >
> > > - *M* include/GCanvas.h
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-0> (2)
> > > - *M* include/TCal.h
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-1> (3)
> > > - *M* include/TEnergyCal.h
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-2> (7)
> > > - *M* include/TNucleus.h
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-3> (2)
> > > - *M* libraries/GROOT/GCanvas.cxx
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-4> (72)
> > > - *M* libraries/TGRSIAnalysis/TCal/TCal.cxx
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-5> (16)
> > > - *M* libraries/TGRSIAnalysis/TCal/TEnergyCal.cxx
> > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-6> (47)
> > >
> > > Patch Links:
> > >
> > > - https://github.com/pcbend/GRSISort/pull/286.patch
> > > - https://github.com/pcbend/GRSISort/pull/286.diff
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/pcbend/GRSISort/pull/286>.
> > >
> > —
> > Reply to this email directly or view it on GitHub <
> https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040476>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040661>.
>

r3dunlop commented:
Yep, that sounds good. We should also make a standard. i.e. Capital letters are for follow up input things…etc I’m not sure what I’m talking about, but we need standards!

> On Apr 24, 2015, at 4:25 PM, pcbend <notifications@github.com> wrote:
> 
> We need to make a new rule; no adding keyboard commands with out updating
> the wiki! :)
> 
> https://github.com/pcbend/GRSISort/wiki/Interactive-Analysis
> 
> I even made a link on the main page.
> 
> On Fri, Apr 24, 2015 at 3:20 PM, Ryan Dunlop <notifications@github.com>
> wrote:
> 
> > f for fit is fine to me. I’ll do it on my end.
> >
> > > On Apr 24, 2015, at 3:19 PM, pcbend <notifications@github.com> wrote:
> > >
> > > I was going to use p for 2d project... how would you fell about switching
> > > to f for find or s for search? or i need to find a new letter for
> > project!
> > >
> > > On Fri, Apr 24, 2015 at 2:57 PM, Ryan Dunlop <notifications@github.com>
> > > wrote:
> > >
> > > > Use 'p' to fit a TPeak...It's pretty boss....
> > > > ------------------------------
> > > > You can view, comment on, or merge this pull request online at:
> > > >
> > > > https://github.com/pcbend/GRSISort/pull/286
> > > > Commit Summary
> > > >
> > > > - Nucleus loads into EnergyCal
> > > > - Can change energy calibration points
> > > > - Added TPeak to Groot
> > > >
> > > > File Changes
> > > >
> > > > - *M* include/GCanvas.h
> > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-0> (2)
> > > > - *M* include/TCal.h
> > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-1> (3)
> > > > - *M* include/TEnergyCal.h
> > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-2> (7)
> > > > - *M* include/TNucleus.h
> > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-3> (2)
> > > > - *M* libraries/GROOT/GCanvas.cxx
> > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-4> (72)
> > > > - *M* libraries/TGRSIAnalysis/TCal/TCal.cxx
> > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-5> (16)
> > > > - *M* libraries/TGRSIAnalysis/TCal/TEnergyCal.cxx
> > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-6> (47)
> > > >
> > > > Patch Links:
> > > >
> > > > - https://github.com/pcbend/GRSISort/pull/286.patch
> > > > - https://github.com/pcbend/GRSISort/pull/286.diff
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <https://github.com/pcbend/GRSISort/pull/286>.
> > > >
> > > —
> > > Reply to this email directly or view it on GitHub <
> > https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040476>.
> >
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040661>.
> >
> —
> Reply to this email directly or view it on GitHub <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96054199>.
>

pcbend commented:
So, I just pushed an updated version. I moved peak fit from 'p' to 'f' and
'F'.  I also changed TPeakFit to be quieter if the "Q" is passed to the
fit.  Right now 'f' does the fit, but doesn't print anything (unless
something goes wrong in the fit!) while "F" gives the full fit result.  My
reasoning behind this is, most of the time we do not care about every
detail of the fit and are really only concerned with the fact that it work,
what the centroid and area are.

At the end of the GCanvas::PeakFitQ, I have an array of variable param and
error (which are the errors for the param values).  I think we should turn
them into a simple report like I did for the crappy gaus fit example... I
haven't added the area yet, most because I am uncertain how the beta
variable effects the area/area-error (if it does at all) .  If anyone wants
to figure that out, it would be great.


On Fri, Apr 24, 2015 at 5:04 PM, Ryan Dunlop <notifications@github.com>
wrote:

> Yep, that sounds good. We should also make a standard. i.e. Capital
> letters are for follow up input things…etc I’m not sure what I’m talking
> about, but we need standards!
>
> > On Apr 24, 2015, at 4:25 PM, pcbend <notifications@github.com> wrote:
> >
> > We need to make a new rule; no adding keyboard commands with out updating
> > the wiki! :)
> >
> > https://github.com/pcbend/GRSISort/wiki/Interactive-Analysis
> >
> > I even made a link on the main page.
> >
> > On Fri, Apr 24, 2015 at 3:20 PM, Ryan Dunlop <notifications@github.com>
> > wrote:
> >
> > > f for fit is fine to me. I’ll do it on my end.
> > >
> > > > On Apr 24, 2015, at 3:19 PM, pcbend <notifications@github.com>
> wrote:
> > > >
> > > > I was going to use p for 2d project... how would you fell about
> switching
> > > > to f for find or s for search? or i need to find a new letter for
> > > project!
> > > >
> > > > On Fri, Apr 24, 2015 at 2:57 PM, Ryan Dunlop <
> notifications@github.com>
> > > > wrote:
> > > >
> > > > > Use 'p' to fit a TPeak...It's pretty boss....
> > > > > ------------------------------
> > > > > You can view, comment on, or merge this pull request online at:
> > > > >
> > > > > https://github.com/pcbend/GRSISort/pull/286
> > > > > Commit Summary
> > > > >
> > > > > - Nucleus loads into EnergyCal
> > > > > - Can change energy calibration points
> > > > > - Added TPeak to Groot
> > > > >
> > > > > File Changes
> > > > >
> > > > > - *M* include/GCanvas.h
> > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-0> (2)
> > > > > - *M* include/TCal.h
> > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-1> (3)
> > > > > - *M* include/TEnergyCal.h
> > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-2> (7)
> > > > > - *M* include/TNucleus.h
> > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-3> (2)
> > > > > - *M* libraries/GROOT/GCanvas.cxx
> > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-4> (72)
> > > > > - *M* libraries/TGRSIAnalysis/TCal/TCal.cxx
> > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-5> (16)
> > > > > - *M* libraries/TGRSIAnalysis/TCal/TEnergyCal.cxx
> > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-6> (47)
> > > > >
> > > > > Patch Links:
> > > > >
> > > > > - https://github.com/pcbend/GRSISort/pull/286.patch
> > > > > - https://github.com/pcbend/GRSISort/pull/286.diff
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <https://github.com/pcbend/GRSISort/pull/286>.
> > > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub <
> > > https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040476>.
> > >
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040661>.
> > >
> > —
> > Reply to this email directly or view it on GitHub <
> https://github.com/pcbend/GRSISort/pull/286#issuecomment-96054199>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96064894>.
>

r3dunlop commented:
I'm not sure what you mean. Tpeak has an area that depends on beta and is stored within the tpeak. It can be accessed with a simple getter, along with its uncertainty. We should also start thinking about a way to save fits via groot. The version of tpeak I put in overwrote itself on every fit. 

> On Apr 24, 2015, at 5:44 PM, pcbend <notifications@github.com> wrote:
> 
> So, I just pushed an updated version. I moved peak fit from 'p' to 'f' and
> 'F'. I also changed TPeakFit to be quieter if the "Q" is passed to the
> fit. Right now 'f' does the fit, but doesn't print anything (unless
> something goes wrong in the fit!) while "F" gives the full fit result. My
> reasoning behind this is, most of the time we do not care about every
> detail of the fit and are really only concerned with the fact that it work,
> what the centroid and area are.
> 
> At the end of the GCanvas::PeakFitQ, I have an array of variable param and
> error (which are the errors for the param values). I think we should turn
> them into a simple report like I did for the crappy gaus fit example... I
> haven't added the area yet, most because I am uncertain how the beta
> variable effects the area/area-error (if it does at all) . If anyone wants
> to figure that out, it would be great.
> 
> 
> On Fri, Apr 24, 2015 at 5:04 PM, Ryan Dunlop <notifications@github.com>
> wrote:
> 
> > Yep, that sounds good. We should also make a standard. i.e. Capital
> > letters are for follow up input things…etc I’m not sure what I’m talking
> > about, but we need standards!
> >
> > > On Apr 24, 2015, at 4:25 PM, pcbend <notifications@github.com> wrote:
> > >
> > > We need to make a new rule; no adding keyboard commands with out updating
> > > the wiki! :)
> > >
> > > https://github.com/pcbend/GRSISort/wiki/Interactive-Analysis
> > >
> > > I even made a link on the main page.
> > >
> > > On Fri, Apr 24, 2015 at 3:20 PM, Ryan Dunlop <notifications@github.com>
> > > wrote:
> > >
> > > > f for fit is fine to me. I’ll do it on my end.
> > > >
> > > > > On Apr 24, 2015, at 3:19 PM, pcbend <notifications@github.com>
> > wrote:
> > > > >
> > > > > I was going to use p for 2d project... how would you fell about
> > switching
> > > > > to f for find or s for search? or i need to find a new letter for
> > > > project!
> > > > >
> > > > > On Fri, Apr 24, 2015 at 2:57 PM, Ryan Dunlop <
> > notifications@github.com>
> > > > > wrote:
> > > > >
> > > > > > Use 'p' to fit a TPeak...It's pretty boss....
> > > > > > ------------------------------
> > > > > > You can view, comment on, or merge this pull request online at:
> > > > > >
> > > > > > https://github.com/pcbend/GRSISort/pull/286
> > > > > > Commit Summary
> > > > > >
> > > > > > - Nucleus loads into EnergyCal
> > > > > > - Can change energy calibration points
> > > > > > - Added TPeak to Groot
> > > > > >
> > > > > > File Changes
> > > > > >
> > > > > > - *M* include/GCanvas.h
> > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-0> (2)
> > > > > > - *M* include/TCal.h
> > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-1> (3)
> > > > > > - *M* include/TEnergyCal.h
> > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-2> (7)
> > > > > > - *M* include/TNucleus.h
> > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-3> (2)
> > > > > > - *M* libraries/GROOT/GCanvas.cxx
> > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-4> (72)
> > > > > > - *M* libraries/TGRSIAnalysis/TCal/TCal.cxx
> > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-5> (16)
> > > > > > - *M* libraries/TGRSIAnalysis/TCal/TEnergyCal.cxx
> > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-6> (47)
> > > > > >
> > > > > > Patch Links:
> > > > > >
> > > > > > - https://github.com/pcbend/GRSISort/pull/286.patch
> > > > > > - https://github.com/pcbend/GRSISort/pull/286.diff
> > > > > >
> > > > > > —
> > > > > > Reply to this email directly or view it on GitHub
> > > > > > <https://github.com/pcbend/GRSISort/pull/286>.
> > > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub <
> > > > https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040476>.
> > > >
> > > > >
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040661>.
> > > >
> > > —
> > > Reply to this email directly or view it on GitHub <
> > https://github.com/pcbend/GRSISort/pull/286#issuecomment-96054199>.
> >
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96064894>.
> >
> —
> Reply to this email directly or view it on GitHub.
>

pcbend commented:
ok... well i clearly didn't look at it carefully enough.  The version using
just the f function adds a new fit every time to the histogram.  I agree,
we should have a way to print the fits, this is pretty easy but we should
also have away to remove selected fits... this is a bit harder.  n still
removes all fits.

On Fri, Apr 24, 2015 at 7:13 PM, Ryan Dunlop <notifications@github.com>
wrote:

> I'm not sure what you mean. Tpeak has an area that depends on beta and is
> stored within the tpeak. It can be accessed with a simple getter, along
> with its uncertainty. We should also start thinking about a way to save
> fits via groot. The version of tpeak I put in overwrote itself on every
> fit.
>
> > On Apr 24, 2015, at 5:44 PM, pcbend <notifications@github.com> wrote:
> >
> > So, I just pushed an updated version. I moved peak fit from 'p' to 'f'
> and
> > 'F'. I also changed TPeakFit to be quieter if the "Q" is passed to the
> > fit. Right now 'f' does the fit, but doesn't print anything (unless
> > something goes wrong in the fit!) while "F" gives the full fit result.
> My
> > reasoning behind this is, most of the time we do not care about every
> > detail of the fit and are really only concerned with the fact that it
> work,
> > what the centroid and area are.
> >
> > At the end of the GCanvas::PeakFitQ, I have an array of variable param
> and
> > error (which are the errors for the param values). I think we should
> turn
> > them into a simple report like I did for the crappy gaus fit example...
> I
> > haven't added the area yet, most because I am uncertain how the beta
> > variable effects the area/area-error (if it does at all) . If anyone
> wants
> > to figure that out, it would be great.
> >
> >
> > On Fri, Apr 24, 2015 at 5:04 PM, Ryan Dunlop <notifications@github.com>
> > wrote:
> >
> > > Yep, that sounds good. We should also make a standard. i.e. Capital
> > > letters are for follow up input things…etc I’m not sure what I’m
> talking
> > > about, but we need standards!
> > >
> > > > On Apr 24, 2015, at 4:25 PM, pcbend <notifications@github.com>
> wrote:
> > > >
> > > > We need to make a new rule; no adding keyboard commands with out
> updating
> > > > the wiki! :)
> > > >
> > > > https://github.com/pcbend/GRSISort/wiki/Interactive-Analysis
> > > >
> > > > I even made a link on the main page.
> > > >
> > > > On Fri, Apr 24, 2015 at 3:20 PM, Ryan Dunlop <
> notifications@github.com>
> > > > wrote:
> > > >
> > > > > f for fit is fine to me. I’ll do it on my end.
> > > > >
> > > > > > On Apr 24, 2015, at 3:19 PM, pcbend <notifications@github.com>
> > > wrote:
> > > > > >
> > > > > > I was going to use p for 2d project... how would you fell about
> > > switching
> > > > > > to f for find or s for search? or i need to find a new letter
> for
> > > > > project!
> > > > > >
> > > > > > On Fri, Apr 24, 2015 at 2:57 PM, Ryan Dunlop <
> > > notifications@github.com>
> > > > > > wrote:
> > > > > >
> > > > > > > Use 'p' to fit a TPeak...It's pretty boss....
> > > > > > > ------------------------------
> > > > > > > You can view, comment on, or merge this pull request online
> at:
> > > > > > >
> > > > > > > https://github.com/pcbend/GRSISort/pull/286
> > > > > > > Commit Summary
> > > > > > >
> > > > > > > - Nucleus loads into EnergyCal
> > > > > > > - Can change energy calibration points
> > > > > > > - Added TPeak to Groot
> > > > > > >
> > > > > > > File Changes
> > > > > > >
> > > > > > > - *M* include/GCanvas.h
> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-0>
> (2)
> > > > > > > - *M* include/TCal.h
> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-1>
> (3)
> > > > > > > - *M* include/TEnergyCal.h
> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-2>
> (7)
> > > > > > > - *M* include/TNucleus.h
> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-3>
> (2)
> > > > > > > - *M* libraries/GROOT/GCanvas.cxx
> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-4>
> (72)
> > > > > > > - *M* libraries/TGRSIAnalysis/TCal/TCal.cxx
> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-5>
> (16)
> > > > > > > - *M* libraries/TGRSIAnalysis/TCal/TEnergyCal.cxx
> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-6>
> (47)
> > > > > > >
> > > > > > > Patch Links:
> > > > > > >
> > > > > > > - https://github.com/pcbend/GRSISort/pull/286.patch
> > > > > > > - https://github.com/pcbend/GRSISort/pull/286.diff
> > > > > > >
> > > > > > > —
> > > > > > > Reply to this email directly or view it on GitHub
> > > > > > > <https://github.com/pcbend/GRSISort/pull/286>.
> > > > > > >
> > > > > > —
> > > > > > Reply to this email directly or view it on GitHub <
> > > > > https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040476>.
>
> > > > >
> > > > > >
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040661>.
>
> > > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub <
> > > https://github.com/pcbend/GRSISort/pull/286#issuecomment-96054199>.
> > >
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96064894>.
> > >
> > —
> > Reply to this email directly or view it on GitHub.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96093439>.
>

pcbend commented:
Now  'N' removes the last fit from the histogram, repeatedly press 'N'
will remove one fit at a time in the reverse order that they where
generated.  'F10' now prints all TPeaks on the histogram to a file
<histname>.fits, if the file already exists - the file is destroyed and
recreated.

I added a chi2 to the printing of TPeaks, but things aren't quite working.
Both the chi2 I added and the farea variable seem to be empty all the
time.  I am not sure why this is.  I am calling it quits for the night.


On Fri, Apr 24, 2015 at 7:28 PM, Peter <pcbend@gmail.com> wrote:

> ok... well i clearly didn't look at it carefully enough.  The version
> using just the f function adds a new fit every time to the histogram.  I
> agree, we should have a way to print the fits, this is pretty easy but we
> should also have away to remove selected fits... this is a bit harder.  n
> still removes all fits.
>
> On Fri, Apr 24, 2015 at 7:13 PM, Ryan Dunlop <notifications@github.com>
> wrote:
>
>> I'm not sure what you mean. Tpeak has an area that depends on beta and is
>> stored within the tpeak. It can be accessed with a simple getter, along
>> with its uncertainty. We should also start thinking about a way to save
>> fits via groot. The version of tpeak I put in overwrote itself on every
>> fit.
>>
>> > On Apr 24, 2015, at 5:44 PM, pcbend <notifications@github.com> wrote:
>> >
>> > So, I just pushed an updated version. I moved peak fit from 'p' to 'f'
>> and
>> > 'F'. I also changed TPeakFit to be quieter if the "Q" is passed to the
>> > fit. Right now 'f' does the fit, but doesn't print anything (unless
>> > something goes wrong in the fit!) while "F" gives the full fit result.
>> My
>> > reasoning behind this is, most of the time we do not care about every
>> > detail of the fit and are really only concerned with the fact that it
>> work,
>> > what the centroid and area are.
>> >
>> > At the end of the GCanvas::PeakFitQ, I have an array of variable param
>> and
>> > error (which are the errors for the param values). I think we should
>> turn
>> > them into a simple report like I did for the crappy gaus fit example...
>> I
>> > haven't added the area yet, most because I am uncertain how the beta
>> > variable effects the area/area-error (if it does at all) . If anyone
>> wants
>> > to figure that out, it would be great.
>> >
>> >
>> > On Fri, Apr 24, 2015 at 5:04 PM, Ryan Dunlop <notifications@github.com>
>>
>> > wrote:
>> >
>> > > Yep, that sounds good. We should also make a standard. i.e. Capital
>> > > letters are for follow up input things…etc I’m not sure what I’m
>> talking
>> > > about, but we need standards!
>> > >
>> > > > On Apr 24, 2015, at 4:25 PM, pcbend <notifications@github.com>
>> wrote:
>> > > >
>> > > > We need to make a new rule; no adding keyboard commands with out
>> updating
>> > > > the wiki! :)
>> > > >
>> > > > https://github.com/pcbend/GRSISort/wiki/Interactive-Analysis
>> > > >
>> > > > I even made a link on the main page.
>> > > >
>> > > > On Fri, Apr 24, 2015 at 3:20 PM, Ryan Dunlop <
>> notifications@github.com>
>> > > > wrote:
>> > > >
>> > > > > f for fit is fine to me. I’ll do it on my end.
>> > > > >
>> > > > > > On Apr 24, 2015, at 3:19 PM, pcbend <notifications@github.com>
>> > > wrote:
>> > > > > >
>> > > > > > I was going to use p for 2d project... how would you fell about
>> > > switching
>> > > > > > to f for find or s for search? or i need to find a new letter
>> for
>> > > > > project!
>> > > > > >
>> > > > > > On Fri, Apr 24, 2015 at 2:57 PM, Ryan Dunlop <
>> > > notifications@github.com>
>> > > > > > wrote:
>> > > > > >
>> > > > > > > Use 'p' to fit a TPeak...It's pretty boss....
>> > > > > > > ------------------------------
>> > > > > > > You can view, comment on, or merge this pull request online
>> at:
>> > > > > > >
>> > > > > > > https://github.com/pcbend/GRSISort/pull/286
>> > > > > > > Commit Summary
>> > > > > > >
>> > > > > > > - Nucleus loads into EnergyCal
>> > > > > > > - Can change energy calibration points
>> > > > > > > - Added TPeak to Groot
>> > > > > > >
>> > > > > > > File Changes
>> > > > > > >
>> > > > > > > - *M* include/GCanvas.h
>> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-0>
>> (2)
>> > > > > > > - *M* include/TCal.h
>> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-1>
>> (3)
>> > > > > > > - *M* include/TEnergyCal.h
>> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-2>
>> (7)
>> > > > > > > - *M* include/TNucleus.h
>> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-3>
>> (2)
>> > > > > > > - *M* libraries/GROOT/GCanvas.cxx
>> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-4>
>> (72)
>> > > > > > > - *M* libraries/TGRSIAnalysis/TCal/TCal.cxx
>> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-5>
>> (16)
>> > > > > > > - *M* libraries/TGRSIAnalysis/TCal/TEnergyCal.cxx
>> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-6>
>> (47)
>> > > > > > >
>> > > > > > > Patch Links:
>> > > > > > >
>> > > > > > > - https://github.com/pcbend/GRSISort/pull/286.patch
>> > > > > > > - https://github.com/pcbend/GRSISort/pull/286.diff
>> > > > > > >
>> > > > > > > —
>> > > > > > > Reply to this email directly or view it on GitHub
>> > > > > > > <https://github.com/pcbend/GRSISort/pull/286>.
>> > > > > > >
>> > > > > > —
>> > > > > > Reply to this email directly or view it on GitHub <
>> > > > > https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040476>.
>>
>> > > > >
>> > > > > >
>> > > > >
>> > > > > —
>> > > > > Reply to this email directly or view it on GitHub
>> > > > > <
>> https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040661>.
>> > > > >
>> > > > —
>> > > > Reply to this email directly or view it on GitHub <
>> > > https://github.com/pcbend/GRSISort/pull/286#issuecomment-96054199>.
>> > >
>> > > >
>> > >
>> > > —
>> > > Reply to this email directly or view it on GitHub
>> > > <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96064894>.
>> > >
>> > —
>> > Reply to this email directly or view it on GitHub.
>> >
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96093439>.
>>
>
>

r3dunlop commented:
It's likely the silly way I wrote fit. I can look into it on Sunday or Monday. 

> On Apr 24, 2015, at 8:38 PM, pcbend <notifications@github.com> wrote:
> 
> Now 'N' removes the last fit from the histogram, repeatedly press 'N'
> will remove one fit at a time in the reverse order that they where
> generated. 'F10' now prints all TPeaks on the histogram to a file
> <histname>.fits, if the file already exists - the file is destroyed and
> recreated.
> 
> I added a chi2 to the printing of TPeaks, but things aren't quite working.
> Both the chi2 I added and the farea variable seem to be empty all the
> time. I am not sure why this is. I am calling it quits for the night.
> 
> 
> On Fri, Apr 24, 2015 at 7:28 PM, Peter <pcbend@gmail.com> wrote:
> 
> > ok... well i clearly didn't look at it carefully enough. The version
> > using just the f function adds a new fit every time to the histogram. I
> > agree, we should have a way to print the fits, this is pretty easy but we
> > should also have away to remove selected fits... this is a bit harder. n
> > still removes all fits.
> >
> > On Fri, Apr 24, 2015 at 7:13 PM, Ryan Dunlop <notifications@github.com>
> > wrote:
> >
> >> I'm not sure what you mean. Tpeak has an area that depends on beta and is
> >> stored within the tpeak. It can be accessed with a simple getter, along
> >> with its uncertainty. We should also start thinking about a way to save
> >> fits via groot. The version of tpeak I put in overwrote itself on every
> >> fit.
> >>
> >> > On Apr 24, 2015, at 5:44 PM, pcbend <notifications@github.com> wrote:
> >> >
> >> > So, I just pushed an updated version. I moved peak fit from 'p' to 'f'
> >> and
> >> > 'F'. I also changed TPeakFit to be quieter if the "Q" is passed to the
> >> > fit. Right now 'f' does the fit, but doesn't print anything (unless
> >> > something goes wrong in the fit!) while "F" gives the full fit result.
> >> My
> >> > reasoning behind this is, most of the time we do not care about every
> >> > detail of the fit and are really only concerned with the fact that it
> >> work,
> >> > what the centroid and area are.
> >> >
> >> > At the end of the GCanvas::PeakFitQ, I have an array of variable param
> >> and
> >> > error (which are the errors for the param values). I think we should
> >> turn
> >> > them into a simple report like I did for the crappy gaus fit example...
> >> I
> >> > haven't added the area yet, most because I am uncertain how the beta
> >> > variable effects the area/area-error (if it does at all) . If anyone
> >> wants
> >> > to figure that out, it would be great.
> >> >
> >> >
> >> > On Fri, Apr 24, 2015 at 5:04 PM, Ryan Dunlop <notifications@github.com>
> >>
> >> > wrote:
> >> >
> >> > > Yep, that sounds good. We should also make a standard. i.e. Capital
> >> > > letters are for follow up input things…etc I’m not sure what I’m
> >> talking
> >> > > about, but we need standards!
> >> > >
> >> > > > On Apr 24, 2015, at 4:25 PM, pcbend <notifications@github.com>
> >> wrote:
> >> > > >
> >> > > > We need to make a new rule; no adding keyboard commands with out
> >> updating
> >> > > > the wiki! :)
> >> > > >
> >> > > > https://github.com/pcbend/GRSISort/wiki/Interactive-Analysis
> >> > > >
> >> > > > I even made a link on the main page.
> >> > > >
> >> > > > On Fri, Apr 24, 2015 at 3:20 PM, Ryan Dunlop <
> >> notifications@github.com>
> >> > > > wrote:
> >> > > >
> >> > > > > f for fit is fine to me. I’ll do it on my end.
> >> > > > >
> >> > > > > > On Apr 24, 2015, at 3:19 PM, pcbend <notifications@github.com>
> >> > > wrote:
> >> > > > > >
> >> > > > > > I was going to use p for 2d project... how would you fell about
> >> > > switching
> >> > > > > > to f for find or s for search? or i need to find a new letter
> >> for
> >> > > > > project!
> >> > > > > >
> >> > > > > > On Fri, Apr 24, 2015 at 2:57 PM, Ryan Dunlop <
> >> > > notifications@github.com>
> >> > > > > > wrote:
> >> > > > > >
> >> > > > > > > Use 'p' to fit a TPeak...It's pretty boss....
> >> > > > > > > ------------------------------
> >> > > > > > > You can view, comment on, or merge this pull request online
> >> at:
> >> > > > > > >
> >> > > > > > > https://github.com/pcbend/GRSISort/pull/286
> >> > > > > > > Commit Summary
> >> > > > > > >
> >> > > > > > > - Nucleus loads into EnergyCal
> >> > > > > > > - Can change energy calibration points
> >> > > > > > > - Added TPeak to Groot
> >> > > > > > >
> >> > > > > > > File Changes
> >> > > > > > >
> >> > > > > > > - *M* include/GCanvas.h
> >> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-0>
> >> (2)
> >> > > > > > > - *M* include/TCal.h
> >> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-1>
> >> (3)
> >> > > > > > > - *M* include/TEnergyCal.h
> >> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-2>
> >> (7)
> >> > > > > > > - *M* include/TNucleus.h
> >> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-3>
> >> (2)
> >> > > > > > > - *M* libraries/GROOT/GCanvas.cxx
> >> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-4>
> >> (72)
> >> > > > > > > - *M* libraries/TGRSIAnalysis/TCal/TCal.cxx
> >> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-5>
> >> (16)
> >> > > > > > > - *M* libraries/TGRSIAnalysis/TCal/TEnergyCal.cxx
> >> > > > > > > <https://github.com/pcbend/GRSISort/pull/286/files#diff-6>
> >> (47)
> >> > > > > > >
> >> > > > > > > Patch Links:
> >> > > > > > >
> >> > > > > > > - https://github.com/pcbend/GRSISort/pull/286.patch
> >> > > > > > > - https://github.com/pcbend/GRSISort/pull/286.diff
> >> > > > > > >
> >> > > > > > > —
> >> > > > > > > Reply to this email directly or view it on GitHub
> >> > > > > > > <https://github.com/pcbend/GRSISort/pull/286>.
> >> > > > > > >
> >> > > > > > —
> >> > > > > > Reply to this email directly or view it on GitHub <
> >> > > > > https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040476>.
> >>
> >> > > > >
> >> > > > > >
> >> > > > >
> >> > > > > —
> >> > > > > Reply to this email directly or view it on GitHub
> >> > > > > <
> >> https://github.com/pcbend/GRSISort/pull/286#issuecomment-96040661>.
> >> > > > >
> >> > > > —
> >> > > > Reply to this email directly or view it on GitHub <
> >> > > https://github.com/pcbend/GRSISort/pull/286#issuecomment-96054199>.
> >> > >
> >> > > >
> >> > >
> >> > > —
> >> > > Reply to this email directly or view it on GitHub
> >> > > <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96064894>.
> >> > >
> >> > —
> >> > Reply to this email directly or view it on GitHub.
> >> >
> >>
> >> —
> >> Reply to this email directly or view it on GitHub
> >> <https://github.com/pcbend/GRSISort/pull/286#issuecomment-96093439>.
> >>
> >
> >
> —
> Reply to this email directly or view it on GitHub.
>
